admin_guide/docker_registry.adoc

```diff
@@ -96,6 +97,113 @@ $ osc exec -p docker-registry-1-da73t find /registry
 <8> This directory contains links to all the layers this repository references
 <9> This file contains a reference to a specific layer that has been linked into this repository via an image
 
+== Securing Registry
+
+=== Overview
+
+You can optionally secure the registry to serve traffic via TLS.
+
+=== Procedure
+
+. Deploy the registry. See <<deploy-registry>> for more information.
++
+. Fetch service IP and port of the registry.
++
+[options="nowrap"]
+----
+$ osc get se docker-registry
+NAME              LABELS                                    SELECTOR                  IP(S)            PORT(S)
+docker-registry   docker-registry=default                   docker-registry=default   172.30.124.220   5000/TCP
+----
++
+. Create server certificates for the registry service IP and docker-registry.default.local
++
+[options="nowrap"]
+----
+$ osadm create-server-cert --signer-cert=ca.crt --signer-key=ca.key --signer-serial=ca.serial.txt --hostnames='docker-registry.default.local,172.30.124.220' --cert=registry.crt --key=registry.key
+----
++
+. Bundle and create secret for registry certificates
++
+[options="nowrap"]
+----
+$ openshift ex bundle-secret registry-secret registry.crt registry.key > secret.json
+$ osc create -f secret.json
+----
++
+. Add secret volume to the registry deployment configuration.
++
+[options="nowrap"]
+----
+$ osc edit dc docker-registry
+...
+volumes:
+  ...
+  - name: secret-volume
+    secret:
+      secretName: registry-secret
+...
+volumeMounts:
+  ...
+  - name: secret-volume
+    mountPath: /etc/secrets/
+    readOnly: true
+----
++
+. Override registry tls config options by adding these environment variables to the registy deployment configuration.
```

Comment:
Enable TLS by adding these environment variables to the deployment configuration: